### ISSUE TITLE
`kubectl create deployment` does not strip off the --image tag

### ISSUE THREAD

**Issue by ahmetb** [kubernetes/kubernetes#43722]
**What keywords did you search in Kubernetes issues before filing this one?** kubectl create "invalid value"  

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Kubernetes version** (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.3", GitCommit:"029c3a408176b55c30846f0faedf56aae5992e9b", GitTreeState:"clean", BuildDate:"2017-02-22T10:12:27Z", GoVersion:"go1.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.2", GitCommit:"08e099554f3c31f6e6f07b448ab3ed78d0520507", GitTreeState:"clean", BuildDate:"1970-01-01T00:00:00Z", GoVersion:"go1.7.1", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**: minikube

**What happened**:

The `--image` flag in `kubectl create deployment` does not strip off the image tag in the `spec.template.spec.containers[0].name` field while making the query

```
$ kubectl create deployment foo --image=gcr.io/ahmetb-162818/hello-node:v1

The Deployment "foo" is invalid: spec.template.spec.containers[0].name: Invalid value: "hello-node:v1": must match the regex [a-z0-9]([-a-z0-9]*[a-z0-9])? (e.g. 'my-name' or '123-abc') 
```

**What you expected to happen**:

Strip off the `:v1` tag in the `--image` and pass `hello-node` to `spec.template.spec.containers[0].name`.

**How to reproduce it** (as minimally and precisely as possible): Run the command listed above.

**Anything else we need to know**:

Here's the API request from -v=10:

```json
{
  "kind": "Deployment",
  "apiVersion": "extensions/v1beta1",
  "metadata": {
    "name": "foo",
    "creationTimestamp": null,
    "labels": {
      "app": "foo"
    }
  },
  "spec": {
    "replicas": 1,
    "selector": {
      "matchLabels": {
        "app": "foo"
      }
    },
    "template": {
      "metadata": {
        "creationTimestamp": null,
        "labels": {
          "app": "foo"
        }
      },
      "spec": {
        "containers": [
          {
            "name": "hello-node:v1",
            "image": "gcr.io/ahmetb-162818/hello-node:v1",
            "resources": {}
          }
        ]
      }
    },
    "strategy": {}
  },
  "status": {}
}
```

**Comment by dixudx** [comment]
Invalid issue, which has already been fixed in #38767.

**Comment by ahmetb** [comment]
Thanks.
